source/Halibut/Transport/PollingClient.cs

```diff
@@ -13,47 +15,66 @@ public class PollingClient : IPollingClient
         readonly ILog log;
         readonly ISecureClient secureClient;
         readonly Uri subscription;
-        readonly Thread thread;
-        readonly CancellationToken cancellationToken;
+        Thread? pollingClientLoopThread;
         bool working;
-        readonly Func<RetryPolicy> createRetryPolicy;
-
-        [Obsolete("Use the overload that provides a logger. This remains for backwards compatibility.")]
-        public PollingClient(Uri subscription, ISecureClient secureClient, Func<RequestMessage, ResponseMessage> handleIncomingRequest, Func<RetryPolicy> createRetryPolicy)
-            : this(subscription, secureClient, handleIncomingRequest, null, createRetryPolicy)
-        {
-        }
 
-        public PollingClient(Uri subscription, ISecureClient secureClient, Func<RequestMessage, ResponseMessage> handleIncomingRequest, ILog log, Func<RetryPolicy> createRetryPolicy)
-            : this(subscription, secureClient, handleIncomingRequest, log, CancellationToken.None, createRetryPolicy)
-        {
-        }
+        Task? pollingClientLoopTask;
+        readonly CancellationTokenSource workingCancellationTokenSource;
```

Comment:
Rather than a bool which is used to eventually stop the thread on dispose I made a linked cancellation token which is cancelled and disposed.